ethicml/evaluators/parallelism.py

```diff
@@ -0,0 +1,99 @@
+"""
+Collection of functions that enable parallelism
+"""
+import asyncio
+from typing import List, Tuple, Union, Coroutine, Any
+
+import pandas as pd
+
+from ethicml.algorithms.utils import DataTuple
+from ethicml.algorithms.inprocess import InAlgorithm, InAlgorithmSync
+
+
+Data = Tuple[DataTuple, DataTuple]
+
+
+async def arrange_in_parallel(
+    algos: List[InAlgorithm], data: Union[Data, List[Data]]
+) -> List[pd.DataFrame]:
+    """Arrange the given algorithms to run (embarrassingly) parallel
+
+    Args:
+        algos: list of algorithms that implement the `run_async` function
+        data: either a single pair of data tuples (train and test) or a list of pairs that has the
+              same length as the list of algorithms
+    Returns:
+        list of the results
+    """
+    coroutines: List[Coroutine[Any, Any, pd.DataFrame]]
+    if isinstance(data, list):
+        assert len(data) == len(algos)
+        coroutines = [algo.run_async(train, test) for algo, (train, test) in zip(algos, data)]
+    else:
+        assert isinstance(data, tuple)
```

Comment:
```suggestion
        assert len(data) == 2
        assert isinstance(data[0], DataTuple)
        assert isinstance(data[1], DataTuple)
```
Would be nice to confirm the input is a tuple of DataTuple as this is pretty close to the user.